src/engine/runner.ts

```diff
@@ -185,31 +206,27 @@ export class Runner {
         atTime: number
     ): AstNodeSnapshot {
         this.profiler.StartProfiling("OvaleBestAction_RecursiveCompute");
-        if (element.result.constant) {
```

Comment:
Why did you remove this code?

Reply:
What has happening was an "if" node with an existing constant result would always return the constant result, even if the condition of the "if" node changed values due to, e.g., checkbox control changing, or items being equipped, etc.

I believe it's because the results aren't being aged correctly, because when I checked the ages of the results, the constant results had very low ages, while the "if" node itself had a very high age.

Reply:
I removed the constant optimization for correctness for now, and then I planned to see how to reintroduce it again.

Reply:
I'm planning on adding some tests to verify the ages of nodes after a script is evaluated.

Reply:
Constant nodes does not care of the age, it's for strings or numbers that should never change. The problem may be that the initial value isn't correctly set when the node is created.

Reply:
Yes, one way or another, there is a bug in the way the AST is evaluated, either in the way that results are stashed in the AST itself or improper aging of results.  Removing that one optimization definitely fixed the problem for now though.  This is a case where we need tests to document the expectation in issue #749 and verify that the right thing is being done by the addon.

Reply:
I fixed the bug in 04c102c842c24e056d68f014142c2a5ec5b57d43, you should be able to revert this code removal.